packages/web-components/src/components/masthead/top-nav-menu.ts

```diff
@@ -21,28 +21,12 @@ const { stablePrefix: ddsPrefix } = ddsSettings;
  */
 @customElement(`${ddsPrefix}-top-nav-menu`)
 class DDSTopNavMenu extends BXHeaderMenu {
-  /**
-   * The trigger button.
-   */
-  @query('[part="trigger"]')
-  private _triggerNode?: HTMLAnchorElement;
-
   /**
    * `true` if this submenu should be in its active state.
    */
   @property({ type: Boolean, reflect: true })
   active = false;
 
-  updated(changedProperties) {
-    super.updated(changedProperties);
-    if (changedProperties.has('active')) {
-      const { active, _triggerNode: triggerNode } = this;
-      if (triggerNode) {
-        triggerNode.setAttribute('data-selected', String(Boolean(active)));
```

Comment:
The `aria-selected` attribute was already addressed here: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/7030. The attribute was changed to `data-selected` as seen in the code being removed. There's corresponding css styles for the `data-selected` attribute so that it works for both react and WC.

Unless there's an issue with `data-selected`, I think we should keep it and remove the `:host(#{$dds-prefix}-megamenu-top-nav-menu)[active] a::after` styles as it would be redundant.

Reply:
Ah, right you are. I hadn't updated my branch from upstream in a little while, so I didn't see that this was already taken care of! Fixed in the latest commit.